gnucash/gnome/assistant-stock-transaction.cpp

```diff
@@ -0,0 +1,1151 @@
+/********************************************************************\
+ * assistant-stock-transaction.c -- stock assistant for GnuCash     *
+ * Copyright (C) 2022 Christopher Lam                               *
+ *                                                                  *
+ * This program is free software; you can redistribute it and/or    *
+ * modify it under the terms of the GNU General Public License as   *
+ * published by the Free Software Foundation; either version 2 of   *
+ * the License, or (at your option) any later version.              *
+ *                                                                  *
+ * This program is distributed in the hope that it will be useful,  *
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of   *
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the    *
+ * GNU General Public License for more details.                     *
+ *                                                                  *
+ * You should have received a copy of the GNU General Public License*
+ * along with this program; if not, contact:                        *
+ *                                                                  *
+ * Free Software Foundation           Voice:  +1-617-542-5942       *
+ * 51 Franklin Street, Fifth Floor    Fax:    +1-617-542-2652       *
+ * Boston, MA  02110-1301,  USA       gnu@gnu.org                   *
+\********************************************************************/
+
+#include <config.h>
+
+#include <gtk/gtk.h>
+#include <glib/gi18n.h>
+#include <vector>
+#include <string>
+
+extern "C" {
+#include "Transaction.h"
+#include "engine-helpers.h"
+#include "dialog-utils.h"
+#include "assistant-stock-transaction.h"
+#include "gnc-account-sel.h"
+#include "gnc-amount-edit.h"
+#include "gnc-component-manager.h"
+#include "gnc-date-edit.h"
+#include "gnc-tree-view-account.h"
+
```

Comment:
Needs a QofLoModule decl e.g.:
```
static QofLogModule log_module = GNC_MOD_ASSISTANT;
```
There are a bunch of GNC_MOD_FOO declared in libgnucash/engine/gnc-engine.h or you can inherit `gnc.gui` from CMakeLists.txt by using `G_LOG_DOMAIN`.